### ISSUE TITLE
HTTP APIs for avalon/pyblish (aport)

### ISSUE THREAD

**Issue by jezscha** [getavalon/core#367]
- Aport is abbreviation for Avalon-port
- it is run as server on `http://localhost:4242` and ip and port could be changed in environment variables
- it is using simple HTTP APIs. Example:  `http://localhost:4242/<python_module>/<module_function>?<argument1>=<value1>?<argument2>=<value2>`
- It could be better connected to javascript, and Extended Scrpt used in Adobe CC so any of their software could be integrated.
- a prototype we have and testing at the moment is able to:
    - **set context** - set avalon session into particular context and host application
    - **deregister pyblish plugin directory paths** - remove all paths from PUBLISH_PATH
    - **register pyblish plugin directory paths** - adding path for host application plugins into PUBLISH_PATH
    - **publish from input json file** - it will open pyblish qml and load instances from json file
- Aport could be run as part of the Avalon Tray menu and it could be waiting for any host to submit new publishing subset
- 

Example of use from adobe Premiere Pro:
![image](https://user-images.githubusercontent.com/40640033/51323822-8cf0f280-1a69-11e9-8540-e6c1bd17518f.png)

**Comment by tokejepsen** [comment]
Dunno if you guys had a look at this? https://github.com/pyblish/pyblish-aftereffects/pull/1

How would we interact with the Adobe apps from actions?

**Comment by jezscha** [comment]
> Dunno if you guys had a look at this? [pyblish/pyblish-aftereffects#1](https://github.com/pyblish/pyblish-aftereffects/pull/1)

Yes we were considering this, but as we were about to implement something more avalon centric and universal for any host, this would not allow us to do so

> How would we interact with the Adobe apps from actions?

Thanks for the question! yes and the way it is done is proposed in #368. But basically. It will run bat script with python host wrapper which will set all environment variables, synchronize a extension into correct adobe respected folder and then run the host application. Python process will end of it could be waiting. But as it is all done from extension to Aport communication, the python wrapper process could exit.

**Comment by mottosso** [comment]
It's very unclear what this is doing.

> `http://localhost:4242/<python_module>/<module_function>?<argument1>=<value1>?<argument2>=<value2>`

Is the idea to facilitate calls to arbitrary Python from JavaScript? It sounds like we need to narrow the scope of this somewhat, has it's scope expanded beyond Premiere and Pyblish? Is it still being used? What is the current state of it?

**Comment by jasperges** [comment]
@jezscha @mkolar are you using this in production? Is it working well?

**Comment by mkolar** [comment]
we are using it at one site. However we didn't have capacity to spend some good time on it, so it's a bit temperamental and not very stable.

**Comment by jasperges** [comment]
Thanks for the info. At least for the moment I won't spend time investigating this. :)
